### ISSUE TITLE
Misleading documentation for "protected-access / W0212"

### ISSUE THREAD

**Issue by clo-vis** [pylint-dev/pylint#9511]
### Bug description

According to the documentation, the message 
"Access to a protected member %s of a client class" 
is issued 
"when a protected member (…) is access outside the class or a descendant of the class where it's defined."
(https://pylint.readthedocs.io/en/latest/user_guide/messages/warning/protected-access.html#protected-access-w0212)

In
```python
class Blah:
    def __init__(self):
        self.index = 1

    def _blah(self) -> None:
        print(self.index)

    def blah(self) -> None:
        self._blah()
        Blah()._blah()

    @staticmethod
    def static_blah() -> None:
        Blah()._blah()
```
The method _blah() is accessED inside the class where it is defined, but pylint issues the message anyway.

### Command used

```shell
pylint blah.py
```

### Pylint output

```python
blah.py:10:8: W0212: Access to a protected member _blah of a client class (protected-access)
blah.py:14:8: W0212: Access to a protected member _blah of a client class (protected-access)
```

### Expected behavior

The description should be changed to reflect the actual behavior. 

### Pylint version

```shell
pylint 3.1.0
astroid 3.1.0
Python 3.11.6 (tags/v3.11.6:8b6ee5b, Oct  2 2023, 14:57:12) [MSC v.1935 64 bit (AMD64)]
```

**Comment by DanielNoord** [comment]
This is correct. You're instantiating a new `Blah()`. This new instance is not "owned" by the instance of `Blah` for which you are calling `static_blah`. Thus, you are indeed accessing a protected member of an instance that is not owned by the callee.

**Comment by clo-vis** [comment]
I didn't say pylint's behavior is incorrect, I said **the documentation** is misleading. The documentation does not say anything about ownership; it says the message is issued if the member "is access [sic] outside the class".

**Comment by Pierre-Sassoulas** [comment]
@clo-vis do you mind opening a PR for it ? The related code is here: 

https://github.com/pylint-dev/pylint/blob/178491e295a0721681a526a622e01b4c01c56b08/pylint/checkers/classes/class_checker.py#L523-L529

**Comment by clo-vis** [comment]
I wouldn't know how to do it **_right_** because I don't know when pylint actually issues the message.

For example, I get the same warning for
```python
import my_module
print(my_module._MyPrivateClass())
```
But I don't know if it is correct to call _my_module_ a client class and _\_MyPrivateClass_ a protected class member.

And I don't get any warning for
```python
from my_module import _MyPrivateClass
print(_MyPrivateClass())
```
Am I not accessing a "protected member" of a "client class" again, just with another syntax?

And for
```python 
class Blah:
    def _replace(a, b): ...
    def _replace_internal(a, b): ...
```
_Blah().\_replace_internal(1, 2)_ generates a warning, but _Blah().\_replace(1, 2)_ does not. Are members called _replace special somehow?

And... and... and...

**Comment by clo-vis** [comment]
The shortest I could come up with: 

> Used when a class or module member whose name begins with an underscore (and is not a dunder name) is accessed in certain places, **for example** outside the class or a descendant of the class where it's defined (with a few exceptions).

- I removed the word _protected_ because names starting with double underscore are also flagged by W0212, and other pylint warnings (such as W0238) call such names "private", not "protected".
- I changed "class member" to "class or module member" (though "module _member_" is not really correct).
- I mentioned dunder names because  W0212 ignores them, even though they begin with an underscore.
- I added "for example" to make it clear that the message is also emitted in other cases.
- I added "with a few exceptions" to cover the cases when no warning is emitted even though the access occurs outside the class/class descendant/module.

**Comment by DanielNoord** [comment]
Is "in certain places, for example outside the class or a descendant of the class where it's defined (with a few exceptions)" really better than "outside the class or a descendant of the class where it's defined."?
